Extra Cloud poller info (#3552)
centreon/centreon-documentation: cloud/installation/poller-troubleshoot.md

## What's inside a poller?

A poller basically consists of two services:

* **Gorgone** (the **gorgoned** process) is responsible for managing the monitoring configuration. It receives and applies configurations on the poller when a user [deploys the configuration](../monitoring/monitoring-servers/deploying-a-configuration.md) for this poller from the Centreon Cloud interface.

* **centengine** or **centreon-engine**/Engine is responsible for scheduling and executing checks on monitored resources and for sending check results to the Centreon Cloud platform.

Both of them are connected to services hosted by Centreon.

## Installation issues

Check the following log file for any errors (located in the folder created by the installation script in the directory where the installation script was executed).

```shell
less ./log/register-poller.log
```

## Monitoring changes are not taken into account when I deploy the configuration

Typically, checks are not executed. Perform the following checks: if a step has worked for you, then go to the next step.

### Step 1: Test Gorgone's Connection to the platform

> Check that Gorgone can connect to your Centreon Cloud platform.

Author: ```suggestion
Check that Gorgone can connect to your Centreon Cloud platform (you may have to install the `nc` command: on EL, use `dnf install nc` and on Debian use `apt install netcat-traditional`).
```

Reviewer: It's ok for the remarks

Once you have set the log level to info or debug, this INFO message should appear in Gorgone logs:

```shell
INFO - [pullwss] websocket connected
```

If you get another message, this means that there is a problem with the network configuration or an authentication problem.

### Step 6: Check that the deploy command works

Check the log messages generated by Gorgone while the configuration is being deployed. The log should contain the following lines, which mean that the poller has received the configuration.

```shell
2024-09-13 14:28:50 - INFO - [action] Copy processing - Received chunk for '/etc/centreon-engine//'
2024-09-13 14:28:51 - INFO - [action] Copy processing - Copy to '/etc/centreon-engine//' finished successfully
2024-09-13 14:28:51 - INFO - [action] Copy processing - Received chunk for '/etc/centreon-broker/'
2024-09-13 14:28:51 - INFO - [action] Copy processing - Copy to '/etc/centreon-broker/' finished successfully
```

If all previous steps were OK but the log does not show that the poller has received the configuration, contact [our support team](https://support.centreon.com/hc/en-us).

### Step 7: Check that configuration files are correctly updated

> Check the modification dates of configuration files: if you see that the files were not updated when you deployed the configuration, check that the **gorgone** user has rights on this folder.

Reviewer 2: suggestion to explain how to control ACL for gorgone:
```suggestion
Check the modification dates of configuration files: if you see that the files were not updated when you deployed the configuration, check that the **gorgone** user has rights on this folder. The group owner of the folder should be one of `centreon-gorgone` gid and the rights for the group should be `rwx`. 
```

```shell

> ls -l /etc/centreon-engine

Reviewer 3: @Nohzoh: shouldn't this be `ls -ld`?

Reviewer 3: Also, we need to document the fix: explain the owner/permissions for that directory

Reviewer 2: The `ls -l` will display all the configuration files written in the folder, it can be interesting to view them all.
However, we could use `ls -al` to display also the folder itself.
```suggestion
ls -al /etc/centreon-engine
```

```

### Step 8: Check that Engine reloads or restarts correctly

Check Engine's log file according to the method you have selected when deploying the configuration.

1. Open the log file:

```shell

> vi /var/log/centreon-engine/centengine.log

Reviewer 3: ```suggestion
tail -f /var/log/centreon-engine/centengine.log
```

```

2. [Deploy the configuration](../monitoring/monitoring-servers/deploying-a-configuration.md) from the interface using either the **reload** or **restart** method.

3. Check the log file for the following messages:

* If you have selected the **reload** method, the log file should contain:

```shell

> [process] [info] [646] Starting to reload configuration.

Author: retour de JDE: Etape 8 : "je n’ai pas les memes messages (ce n’est pas du 646)"

Reviewer: waiting answer ?

Reviewer 2: this is not relevant
the important thing is the message that the configuration is being reloaded

> [process] [info] [646] Starting to reload configuration.

Author: ```suggestion
[process] [info] [xxx] Starting to reload configuration.
```

(...)

> [process] [info] [646] Reload configuration finished.

Author: ```suggestion
[process] [info] [xxx] Reload configuration finished.
```

```

* If you have selected the **restart** method, the log file should contain:

```shell
[process] [info] [3142] Centreon Engine yy.mm.p starting ...
```

> If it doesn't, check that **gorgoned** has the right to write in `/etc/centreon-engine` and that **centengine** has the right to read in it.

Reviewer 3: This sentence was what step 7 was supposed to check; do we still need it there?

Reviewer 2: Correct.
In this case, there is no clear action on next steps, it depends on other logs messages.
```suggestion
If it doesn't, check errors in logs to get more information about any issue.
```
